src/App.js

```diff
@@ -71,94 +77,59 @@ const App = () => {
+      await addDoc(expensesRef, expense);
 
       if (newExpenseMonth !== selectedMonth) {
         setConfirmMonthModalIsOpen(true);
       }
-
-      setExpenses([...expenses, newExpense]);
-    } catch (error) {
+    } catch (err) {
+      console.log(err);
       setErrorOccurred(true);
     }
 
     setUserAction('add_expense');
     setIsAlertOpen(true);
   };
 
-  const removeExpense = async id => {
-    const url = `${baseURL}/items/${id}.json`;
-    const requestOptions = { method: 'DELETE', headers };
-
-    setConfirmDeleteModalIsOpen(true);
-
+  const removeExpense = async expenseId => {
     try {
-      const res = await fetch(url, requestOptions);
+      const expenseRef = doc(db, 'expenses', expenseId);
+      await deleteDoc(expenseRef);
 
-      // Ensure that the catch block is called before setting local states
-      if (!res.ok) {
-        throw new Error('Something went wrong while deleting the expense!');
+      if (filteredExpenses.length === 1) {
+        setSelectedMonth(currentMonth);
       }
-
-      setExpenses(expenses.filter(expense => expense.id !== id));
-
-      if (filteredExpenses.length === 1) setSelectedMonth(currentMonth);
-      setConfirmDeleteModalIsOpen(false);
-    } catch (error) {
+    } catch (err) {
+      console.log(err);
       setErrorOccurred(true);
-      setConfirmDeleteModalIsOpen(false);
     }
 
+    setConfirmDeleteModalIsOpen(false);
     setUserAction('delete_expense');
     setIsAlertOpen(true);
   };
 
-  const editExpense = async (id, newExpenseData) => {
-    const url = `${baseURL}/items/${id}.json?stash=${token}`;
-    const requestOptions = {
-      method: 'PUT',
-      headers,
-      body: JSON.stringify(newExpenseData),
-    };
-
+  const editExpense = async (expenseId, newExpenseData) => {
     try {
-      const res = await fetch(url, requestOptions);
-      const updatedExpense = await res.json();
-
-      if (!res.ok) {
-        throw new Error('Something went wrong while editing the expense!');
-      }
-
-      const expensesCopy = [...expenses];
-      const indexToEdit = expensesCopy.findIndex(expense => expense.id === id);
-      expensesCopy.splice(indexToEdit, 1, updatedExpense);
-      setExpenses(expensesCopy);
+      const expenseRef = doc(db, 'expenses', expenseId);
+      await updateDoc(expenseRef, newExpenseData);
 
-      // Handle month selection if moving last expense to a different month
-      const updatedExpenseMonth = getUKFormattedDate(updatedExpense.date, {
+      const updatedExpenseMonth = getUKFormattedDate(newExpenseData.date, {
         year: 'numeric',
         month: 'long',
       });
 
-      if (updatedExpenseMonth !== selectedMonth && filteredExpenses.length === 1) {
+      if (
+        updatedExpenseMonth !== selectedMonth &&
+        filteredExpenses.length === 1
+      ) {
         setSelectedMonth(currentMonth);
       }
-    } catch (error) {
+    } catch (err) {
+      console.log(err);
       setErrorOccurred(true);
     }
```

Comment:
very readable. good job. for later, i think it might be nice to move all these persistence methods to another file/class.